clients/src/main/java/org/apache/kafka/clients/CommonClientConfigs.java

```diff
@@ -93,6 +93,10 @@
                                                          + "elapses the client will resend the request if necessary or fail the request if "
                                                          + "retries are exhausted.";
 
+    public static final String CLIENT_DNS_LOOKUP_CONFIG = "client.dns.lookup";
```

Comment:
@edoardocomar Since this PR is using the same config as https://github.com/apache/kafka/pull/4485, it will need to get merged on top of that right?

Reply:
We copied all common files from #4485 (at the same time addressing the points we raised there) so they can be merged in any order.